core/include/kernel/mutex_pm_aware.h

```diff
@@ -17,9 +17,13 @@
  * non-secure world is not operational, for example in fastcall SMC entries
  * of the PSCI services. In such case we cannot take a mutex and we expect
  * the mutex is unlocked. Additionally a spinning lock is attempted to be
- * locked to check the resource access consistency. In case of contention
- * between thread and non-thread access to the resource, core will
- * intentionally panic.
+ * locked to check the resource access consistency.
+ *
+ * In case a thread request the mutex while a non-thread access still holds
+ * it, core will intentionally panic since it is not expected.
+ *
+ * In case a non-thread access request the mutex while it is held by a thread,
+ * core will also panic since it is not an expected situation.
```

Comment:
Please add the third case with non-threaded access while another core holds the spinlock.

Reply:
Ok. Comments addressed.